### ISSUE TITLE
False Datetime Validation after Upgrade 3.8.2 -> 4.3.0

### ISSUE THREAD

**Issue by martinwurst1** [FirelyTeam/firely-net-sdk#2313]
After upgrading from Hl7.Fhir.R4 3.8.2 to 4.3.0 the validation of previously valid bundles failed. I did not change any tar-bundles for validation or the fhir document itself.
The error I receive is "Instance failed constraint correctDate "Begrenzung der Datumsangabe auf 10 Zeichen JJJJ-MM-TT"", which means that the date has to be limited to 10 digits yyyy-mm-dd. However, checking the corresponding entry in the fhir bundle the datetime looks completely fine:

`<extension url="http://hl7.org/fhir/StructureDefinition/condition-assertedDate">			
          <valueDateTime value="2020-02-03" />
  </extension>`

Since the problem only appeared after the update of the validator I assume that it's not a problem of the fhir package.

**Reproduce**:
I uploaded a console application which produces the error:  
https://github.com/martinwurst1/FhirAuConsole

The fhir document is embedded, I added comments to the critical datetime.

**Version used:**
 - FHIR Version: R4
 - Version: 4.3.0

**Comment by ewoutkramer** [comment]
Hi Martin -- I tried to connect the instance data with the profile, but I am a bit stuck:  The error text you have copied here is clearly German, so this looks like an invariant made specifically in the German profiles. However, the instance data you pasted (with the extension) refers to `http://hl7.org/fhir/StructureDefinition/condition-assertedDate`, which is part of the FHIR core extensions. So, somewhere the dots do not connect.

Are you sure the error is really about that extension?

**Comment by mbaltus** [comment]
The newer release of the profiles (version 1.1.0) has a corrected expression by the way, so maybe you can update to using the newer definitions.

**Comment by martinwurst1** [comment]
@mbaltus thanks for the reply, and indeed: In the version 1.1.0 I don't see the error. However, the newer version is active not before June 2023, so I have to be compatible with **both** versions.

**Comment by mbaltus** [comment]
I guess you are compatible with their intent, but their profile is just not correct. So if you want to test and validate your data, any validator will choke on that. Of course, another option is to downgrade to the SDK library that did not give you problems just to get rid of the error.

**Comment by martinwurst1** [comment]
Another option for me would be to change the profile itself. Do I just need to remove the quotes around the url:

`"extension(\http://hl7.org/fhir/StructureDefinition/condition-assertedDate\).valueDateTime.toString().length() = 10"`

**Comment by ewoutkramer** [comment]
No, the character for string separators in FhirPath is the single quote, so it should be "extension('http://hl7.org/fhir/StructureDefinition/condition-assertedDate').valueDateTime.toString().length() = 10".
